### ISSUE TITLE
Doesn't play well with another collector

### ISSUE THREAD

**Issue by tomaspinho** [zalando-incubator/kube-metrics-adapter#94]
## Expected Behavior
`kube-metrics-adapter` should behave well in an environment with a different external or custom metrics provider.

## Actual Behavior
`kube-metrics-adapter` will attempt modifying HPAs that have external or custom metrics when it is not configured as a metrics provider for that metric type.

Logs:
```
time="2019-11-26T10:00:23Z" level=info msg="Event(v1.ObjectReference{Kind:\"HorizontalPodAutoscaler\", Namespace:\"development\", Name:\"task-worker\", UID:\"8e7d8d12-0c84-11ea-8744-0a26919e3b5a\", APIVersion:\"autoscaling/v2beta2\", ResourceVersion:\"92543736\", FieldPath:\"\"}): type: 'Warning' reason: 'CreateNewMetricsCollector' Failed to create new metrics collector: no plugin found for {External {aws.sqs.sqs_messages_visible &LabelSelector{MatchLabels:map[string]string{aws_account_name: development,queue_name: tasks,region: eu-west-1,},MatchExpressions:[]LabelSelectorRequirement{},}}}"
```

Event on respective HPA:
```
  Type     Reason                     Age                   From                  Message
  ----     ------                     ----                  ----                  -------
  Warning  CreateNewMetricsCollector  94s (x1947 over 16h)  kube-metrics-adapter  Failed to create new metrics collector: no plugin found for {External {aws.sqs.sqs_messages_visible &LabelSelector{MatchLabels:map[string]string{aws_account_name: development,queue_name: tasks,region: eu-west-1,},MatchExpressions:[]LabelSelectorRequirement{},}}}
```

## Steps to Reproduce the Problem

  1. Install this project only as a Custom Metrics Provider
  2. Install `datadog-cluster-agent` as an External Metrics Provider, for instance. 
  3. Configure a HPA targetting External Metrics

## Specifications

  - Version: v0.0.5
  - Platform: Kubernetes
  - Subsystem: HPA provider

**Comment by tomaspinho** [comment]
Is there a surefire way of introducing logic to the HPA provider to only target HPAs for which the collector is configured for? Or that have any annotations beginning with `metric-config`?

**Comment by mikkeloscar** [comment]
What is the actual impact you observe? Does it just print errors because it can't enable a collector for the HPA, or does it cause actual issues?

If it's just an error we can maybe maybe it less verbose?

**Comment by tomaspinho** [comment]
As it fails to create the collector, the errors get logged and inserted as events on the HPA. We're seeing thousands of such events, it's unnecessary noise. But there's no other impact.

**Comment by mikkeloscar** [comment]
Ah ok, I see what you mean now. Yeah, that is annoying if you don't intend to use it for this.

In the general case I think the event/log is still useful if you are only running the kube-metrics-adapter in your cluster, but it would probably make sense to add a flag for disabling reporting HPAs which doesn't match any collector plugin. What do you think?

**Comment by tomaspinho** [comment]
Yeah, that's the kind of thing I'd be willing to implement. What do you think about the two ways I suggested above?

**Comment by mikkeloscar** [comment]
> Yeah, that's the kind of thing I'd be willing to implement. What do you think about the two ways I suggested above?

The best way would be to check if the error here: https://github.com/zalando-incubator/kube-metrics-adapter/blob/120950078c9b21bcee0dae7a1011b94d1e25ab18/pkg/provider/hpa.go#L165 is indicating that no plugin matched the HPA and then just not log the error/emit the event if you have this disabled via a flag.

Does that make sense?

**Comment by tomaspinho** [comment]
PR for this opened

**Comment by arjunrn** [comment]
Issue fixed with #95
